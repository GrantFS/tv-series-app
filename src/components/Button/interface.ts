import { BtnType } from '../../common/types';

export interface ButtonProps {
  href: string;
  name: string;
  type: BtnType;
  classes: string;
}

