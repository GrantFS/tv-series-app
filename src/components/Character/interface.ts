export interface CharacterProps {
  character: Character

}

interface Character {
  image: Image,
  name: string
}

interface Image {
  medium: string;
  original: string;
}


