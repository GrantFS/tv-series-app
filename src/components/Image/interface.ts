import { ActivePhotoFunction } from "../../common/interfaces";

export interface ImageProps {
  image : Image;
  image_key: number;
  setActivePhoto: ActivePhotoFunction;
}

interface Image {
  id: number;
  resolutions: Resolutions;
  type: string;
}

interface Resolutions {
    medium: ImageSize;
    original: ImageSize;
}

interface ImageSize {
  url: string;
}


