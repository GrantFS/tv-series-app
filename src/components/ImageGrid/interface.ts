import { ActivePhotoFunction } from "../../common/interfaces";

export interface ImageGridProps {
  images: Array<Images>;
  setActivePhoto: ActivePhotoFunction;
  extra_classes: string;
}

interface Images {
  [index: number]: { id:number; image_key: number; setActivePhoto: ActivePhotoFunction };
}
