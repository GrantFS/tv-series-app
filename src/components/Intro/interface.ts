export interface IntroProps {
  message: string;
}
