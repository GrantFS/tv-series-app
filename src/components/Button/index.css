.btn {
    display: inline-block;
    line-height: 1.6;
    text-decoration: none;
    border-radius: 4px;
    box-shadow: 0 1px 3px 0 rgba(48, 49, 51, 0.30);
    border: 1px solid transparent;
    padding: 6px 12px;
    font-weight: 500;
    cursor: pointer;
    outline: none;
    margin-bottom: 0;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;
    touch-action: manipulation;
    cursor: pointer;
}

.btn-primary {
    background-color: #3097D1;
    border-color: #2a88bd;
    color: white;
}

.btn-primary:hover {
    background-color: #2579a9;
    border-color: #1f648b;
}

.btn-danger {
    background-color: #a94442;
    border-color: #aa4a24;
    color: white;
}

.btn-danger:hover {
    color: #fff;
    background-color: #954120;
    border-color: #78341a;
}

.btn-success {
    background-color: #2ab27b;
    border-color: #259d6d;
    color: white;
}

.btn-success:hover {
    color: #fff;
    background-color: #20895e;
    border-color: #196c4b;
}

.btn-block {
    display: block;
}