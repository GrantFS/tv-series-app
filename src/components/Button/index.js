import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './index.css';

class Button extends Component {
  
  render() {
    let { href, name, type } = this.props;

    return (
      <div>
        <Link to={href} className={`btn btn-${type}`}>
          {name}
        </Link>
        
      </div>
    );
  }
}

export default Button;