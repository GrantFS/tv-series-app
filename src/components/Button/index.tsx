import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './index.css';
import { ButtonProps } from './interface';

class Button extends Component <ButtonProps> {

  render() {
    let { href, name, type } = this.props;

    return (
      <div>
        <Link data-testid="button" to={href} className={`btn btn-${type}`}>
          {name}
        </Link>

      </div>
    );
  }
}

export default Button;
