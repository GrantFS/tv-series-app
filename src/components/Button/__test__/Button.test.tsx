import React from 'react';
import ReactDOM from 'react-dom';
import Button from '..';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import  TestRenderer from "react-test-renderer";
import { ButtonProps } from '../interface';

test("Renders the correct content", () => {
  const root = document.createElement("div");
  ReactDOM.render(<BrowserRouter><Button href="/" name="Test" /></BrowserRouter>, root);
});

test("Render Save button", () => {
  const { getByTestId } = render(<BrowserRouter><Button name="Save" href="/"></Button></BrowserRouter>)
  expect(getByTestId("button")).toHaveTextContent("Save")
  expect(getByTestId("button")).toHaveAttribute('href', '/');
});

test("Render Click Me button", () => {
  const { getByTestId } = render(<BrowserRouter><Button name="Click Me" href="/"></Button></BrowserRouter>)
  expect(getByTestId("button")).toHaveTextContent("Click Me")
});

test("Render Classes for button", () => {
  const { getByTestId } = render(<BrowserRouter><Button name="Click Me" type="success" href="/"></Button></BrowserRouter>)
  expect(getByTestId("button").classList.contains('btn-success')).toBe(true)
  expect(getByTestId("button").classList.contains('btn')).toBe(true)
});
test("Render href for button", () => {
  const { getByTestId } = render(<BrowserRouter><Button name="Click Me" type="success" href="/test"></Button></BrowserRouter>)
  expect(getByTestId("button")).toHaveAttribute('href', '/test');
});

test("Matches snapshot", () => {
  const tree = TestRenderer.create(<BrowserRouter><Button type="success" name="Click Me" href="/"></Button></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
