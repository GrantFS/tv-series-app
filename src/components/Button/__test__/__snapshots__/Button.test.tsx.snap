// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div>
  <a
    className="btn btn-success"
    data-testid="button"
    href="/"
    onClick={[Function]}
  >
    Click Me
  </a>
</div>
`;
